crates/orchestrator/src/database/mongodb/mod.rs

```diff
@@ -64,17 +65,20 @@ impl MongoDb {
         if result.modified_count == 0 {
             return Err(eyre!("Failed to update job. Job version is likely outdated"));
         }
-        self.update_job_version(current_job).await?;
+        self.post_job_update(current_job).await?;
```

Comment:
i can't recall why we were doing this actually? why don't we just increment the version inside `update`?

Reply:
We actually discussed about this previously. Some mongo issue was there when injecting this inc update in the actual document for updating. That's why at that time we agreed for this approach.

Reply:
Ok, let's create an issue for this and try to tackle it. Because even if we go down this route then I think the two updates should be part of a txn